analyses/cell-type-ewings/cell_type_annotations/SCPCS000490/SCPCL000822_tumor-cell-validation.Rmd

```diff
@@ -0,0 +1,492 @@
+  as.matrix()
+
+ComplexHeatmap::Heatmap(
+  geneset_heatmap,
+  col = circlize::colorRamp2(c(0, 1), colors = c("white", "#00274C")),
+  border = TRUE,
+  ## Row parameters
+  cluster_rows = TRUE,
+  row_title = "",
+  row_title_gp = grid::gpar(fontsize = 12),
+  row_title_side = "left",
+  row_names_side = "left",
+  row_dend_side = "right",
+  row_names_gp = grid::gpar(fontsize = 10),
+  ## Column parameters
+  cluster_columns = TRUE,
+  show_column_names = FALSE,
+  bottom_annotation = annotation
+)
+```
+
+For the most part we see that cells with higher gene set scores are classified as tumor cells. 
+One thing to note is that the `SILIGAN` gene set appears to have some expression in normal cells. 
+However, if you look at the spots where all three gene sets have higher scores, these apply to tumor cells. 
+
+### Validate based on CNAs 
+
+The last thing we will do here is look to see if tumor cells have any of the expected CNAs found in Ewing sarcoma: 
+
+- Gain of Chr8
+- Gain of Chr12
+- Gain of Chr1p
+- Loss of Chr16q
+
+Although these are the most frequent, there are patients who do not have any of these alterations and patients that only have some of these alterations. 
+See [Tirode et al.,](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4264969/) and [Crompton et al.](https://doi.org/10.1158/2159-8290.CD-13-1037). 
+
+We will look specifically at gains and losses in each chromosome across tumor and normal cells. 
+To do this, we will use the output from `InferCNV` since that output separates out gains and losses in a way that makes it easy for plotting. 
+
+The first heatmap looks at the proportion of each chromosome with a CNV that corresponds to an amplification or gain. 
+Here each row is a cell and each column is a chromosome with the color corresponding to the proportion of the chromosome with a gain detected. 
+
+```{r}
+# get matrix of barcodes by chromosome
+gain_cnv_df <- coldata_df |> 
+  dplyr::select(barcodes, starts_with("proportion_scaled_dupli")) |> 
+  unique() |> 
+  tibble::column_to_rownames("barcodes") |> 
+  as.matrix()
+
+# remove extra characters in colnames, proportion_scaled_dupli
+colnames(gain_cnv_df) <- stringr::word(colnames(gain_cnv_df), -1, sep = "_")
+
+# need to make a new annotation, since we want to annotate the rows here 
+annotation <- ComplexHeatmap::rowAnnotation(classification = annotation_df$combined_cnv_classification,
+                                            col = list(classification = c("Tumor" = "#00274C", "Normal" = "#FFCB05")))
+```
+
+
+```{r}
+ComplexHeatmap::Heatmap(
+  gain_cnv_df,
+  col = circlize::colorRamp2(c(0, 1), colors = c("white", "#00274C")),
+  border = TRUE,
+  ## Row parameters
+  cluster_rows = TRUE,
+  row_title = "",
+  row_title_gp = grid::gpar(fontsize = 12),
+  row_title_side = "left",
+  show_row_names = FALSE,
+  row_dend_side = "right",
+  row_names_gp = grid::gpar(fontsize = 10),
+  ## Column parameters
+  cluster_columns = FALSE,
```

Comment:
I see you have this right, but the sorting is still not correct for some reason?